cmd/explain/explain.go

```diff
@@ -0,0 +1,130 @@
+package explain
+
+import (
+	"fmt"
+
+	"github.com/puppetlabs/pdkgo/docs/md"
+	"github.com/puppetlabs/pdkgo/pkg/docs"
+	"github.com/spf13/cobra"
+)
+
+var (
+	docsApi      *docs.Docs
+	listTopics   bool
+	format       string
+	tag          string
+	category     string
+	topic        string
+	// Possibly implement later to enable context aware filtering for tags/categories
+	// depending on which is filtered first?
+	// filteredDocs []docs.MarkdownDoc
+)
+
+func CreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "explain",
+		Short:             "Present documentation about topics",
+		Long:              "Present documentation about various topics, including...",
+		Args:              validateArgCount,
+		ValidArgsFunction: flagCompletion,
+		PreRun:            preExecute,
+		RunE:              execute,
+	}
+
+	dfs := md.GetDocsFS()
+	docsApi = &docs.Docs{
+		DocsFileSystem: &dfs,
+	}
+
+	cmd.Flags().SortFlags = false
+	cmd.Flags().BoolVarP(&listTopics, "list", "l", false, "list available topics")
+
+	cmd.Flags().StringVarP(&format, "format", "f", "human", "display output in human or json format")
+	err := cmd.RegisterFlagCompletionFunc("format", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return []string{"table", "json"}, cobra.ShellCompDirectiveNoFileComp
+	})
+	cobra.CheckErr(err)
+
+	cmd.Flags().StringVarP(&tag, "tag", "t", "", "filter available topics by tag")
+	err = cmd.RegisterFlagCompletionFunc("tag", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		if docsApi.ParsedDocsCache == nil {
+			preExecute(cmd, args)
```

Comment:
I followed the pattern here from the `flagCompletion` in the `new` command without much forethought; I assumed this was necessary since the preExecute may not have already run by the time the completer kicks in.

https://github.com/puppetlabs/pdkgo/blob/6d6cece0abd5412194183e7b677516eb5319d152/cmd/new/new.go#L111-L118